apps/rewards/src/index.ts

```diff
@@ -184,38 +184,51 @@ const getTrades = async (
   }));
 }
 
-const reinvestToken = async (
-  env: Env,
-  provider: any,
-  vault: typeof vaults[0],
-  sellToken: string,
-  amount: BigNumber
-) => {
-  const sellAmountsPerToken = vault.tokenWeights.map((weight: number) => {
-    return amount.mul(weight).div(100);
-  });
+const reinvestVault = async (env: Env, provider: any, vault: typeof vaults[0]) => {
+    if (await shouldSkipReinvest(env, vault.address)) {
+      console.log(`Skipping reinvestment for ${vault.address}, already invested`);
+      return null;
+    }
 
-  const trades = await getTrades(
-    env,
-    sellToken,
-    vault.poolTokens,
-    sellAmountsPerToken,
-  );
+  const tradesPerRewardToken = [];
+  for (const sellToken of vault.rewardTokens) {
+    const amount = await ERC20__factory.connect(sellToken, provider).balanceOf(vault.address);
+    console.log("amout is: ", amount);
+    if (amount.lt(100)) {
+      console.log(`Skipping reinvestment for ${vault.address}: ${sellToken}, 0 reward token`);
+      continue;
+    }
+    const sellAmountsPerToken = vault.tokenWeights.map((weight: number) => {
+      return amount.mul(weight).div(100);
+    });
+
+    const trades = await getTrades(
+      env,
+      sellToken,
+      vault.poolTokens,
+      sellAmountsPerToken,
+    );
+    tradesPerRewardToken.push(trades);
+  }
+  if (tradesPerRewardToken.length == 0) {
+    console.log(`Found no reward tokens on vault ${vault.address}, skipping.`);
+    return null;
+  }
 
   const data = TreasuryManager__factory.connect(
```

Comment:
Can you run a `staticCall` against this method and get the return values? It should return the pool claim amounts that each reinvestment will generate. Then you can use those values and discount them by 1% as inputs here.